test/test_array/array.h

```diff
@@ -21,11 +21,14 @@ class array : public teca_dataset
     TECA_DATASET_PROPERTY(std::vector<size_t>, extent)
     TECA_DATASET_PROPERTY(std::string, name)
 
-
     // return a unique string identifier
     std::string get_class_name() const override
     { return "array"; }
 
+    // return an integer identifier uniquely naming the dataset type
+    int get_type_code() const override
+    { return 1; }
```

Comment:
This appears to be the typecode for a TECA table, according to `data/teca_dataset_util.h` -- is this correct?  

If not, could you please add some commentary here describing why the number 1 is used and why the typecode isn't determined at runtime (e.g., from the `teca_dataset_tt` struct)? I might not be understanding what this test_array code is used for, and I'm not immediately able to tell from looking at the source.

Reply:
this is a prototyping code that predates the rest of teca. it shouldn't be referenced from the inside of teca. I'll add a readme file to this directory to make this clear.